src/types/verifiedAddress.ts

```diff
@@ -0,0 +1,36 @@
+import { Address } from './publicAPI'
+
+/**
+ * Object containing address information alongside signatures.
+ */
+// eslint-disable-next-line import/no-unused-modules -- will disable in next PR
+export interface VerifiedAddress {
+  readonly payload: VerifiedAddressPayload
+  readonly signatures: VerifiedAddressSignature[]
```

Comment:
I think we would want to make the array readonly as well, right?

```suggestion
  readonly signatures: readonly VerifiedAddressSignature[]
```